app/post.go

```diff
@@ -717,6 +719,134 @@ func (a *App) GetPostsAroundPost(postId, channelId string, offset, limit int, be
 	return result.Data.(*model.PostList), nil
 }
 
+func (a *App) GetPostAfterTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	result := <-a.Srv.Store.Post().GetPostAfterTime(channelId, time)
+	if result.Err != nil {
+		return nil, result.Err
+	}
+
+	return result.Data.(*model.Post), nil
+}
+
+func (a *App) GetPostBeforeTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	result := <-a.Srv.Store.Post().GetPostBeforeTime(channelId, time)
+	if result.Err != nil {
+		return nil, result.Err
+	}
+
+	return result.Data.(*model.Post), nil
+}
+
+func (a *App) GetNextPostIdFromPostList(postList *model.PostList) string {
+	if len(postList.Order) > 0 {
+		firstPostId := postList.Order[0]
+		firstPost := postList.Posts[firstPostId]
+		nextPost, err := a.GetPostAfterTime(firstPost.ChannelId, firstPost.CreateAt)
+		if err != nil {
+			mlog.Error("GetNextPostIdFromPostList: failed in getting next post", mlog.Any("err", err))
```

Comment:
re: `mlog.Error`, there's a topic queued in dev meeting.  Not sure what would be the alternative.  I can probably add it as a follow change if necessary.

Reply:
yes, maybe change to a warning? and I think the parameter should be `err.Error()`